### PR TITLE
[inductor][eager] Resolve which distribution provides the importable Triton module

### PR DESCRIPTION
**Pull request by naromero77amd**
**Context:** split out of #194131, and addresses @slayton58's comment that the lookup returns `triton_rocm`'s version instead of `triton`'s and that "we should be careful for the event that" the two do not align.

### The problem

Native DSL ops gate on the Triton version, read from distribution metadata. `version("triton")` resolves by distribution *name* and never checks which files that distribution installed:

- Normalization never strips a suffix, so `triton-rocm` and `fbtriton` miss. The ops then silently do not register on an install where `import triton` works.
- A provider swap leaves the previous dist-info behind, still reporting a version for a module it no longer owns (`TRITON_DISTRIBUTIONS` in `tools/torchtlx/dev.py`).

### The fix

Collect versions from known distribution names before scanning `sys.path`. A single candidate is used directly. When multiple distributions report versions, compare the hash recorded for the resolved `triton` module with its bytes on disk; scan only if the known candidates remain unresolved.

- `find_spec` locates the module without importing it.
- Unlisted names still resolve through the scan, and names are compared using metadata normalization.
- A missing RECORD or hash and an editable-install shim RECORD are undecidable; an empty RECORD is evidence of non-ownership.
- One unreadable provider does not discard providers listed after it. Irreducible ties warn and preserve the prior preference for `triton`.

A lone candidate is intentionally not verified, preserving `main`'s behavior and avoiding RECORD work on the common path. Consequently, a lone stale known name can still hide an unlisted provider.

Also fixed: malformed metadata could raise during `import torch` and was logged only at `debug`; it is now caught and reported at `warning`.

### Cost

Against `main`'s single `version("triton")`, the common path measured **0.27 ms -> 0.76 ms**, once per process behind `functools.cache`. A known same-path collision resolves in about **1.7 ms**. The full 419 ms `sys.path` scan is reserved for an unknown provider or unresolved ambiguity.

### Test Plan

```
python test/python_native/test_native_dsl_ops.py
python test/test_public_bindings.py TestPublicBindings.test_correct_module_names
lintrunner torch/_native/triton_utils.py test/python_native/test_native_dsl_ops.py
```

Tests cover known and unlisted names, same-path RECORD collisions with different hashes, editable installs, missing/empty metadata, unreadable providers, normalized names, ambiguity fallback, and the version gate boundaries. The collision was also reproduced in isolated virtual environments with two actual wheels installed in both orders; the lookup selected the wheel whose bytes were live each time.

Made with [Cursor](https://cursor.com)

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @pragupta @jerrymannil @xinyazhang

### CODE REVIEW

**Comment by jansel** [comment]
@claude review this

**Comment by jansel** [comment]
@claude review again

**Comment by jansel** [comment]
@claude review again

**Comment by naromero77amd** [comment]
@jansel @eellison Could I get a re-review on this PR?

**Review by jansel — APPROVED** [review]

**Comment by naromero77amd** [comment]
@claude review again. Focus on critical blockers.

**Comment by naromero77amd** [comment]
@claude review again. Focus on critical blockers.

**Comment by naromero77amd** [comment]
@claude review again. Was the first non-block change from your previous review addressed?
